packages/dds/shared-object-base/src/sharedObject.ts

```diff
@@ -3,450 +3,46 @@
  * Licensed under the MIT License.
  */
 
-import { v4 as uuid } from "uuid";
-import { ITelemetryLogger } from "@fluidframework/common-definitions";
-import { assert } from "@fluidframework/common-utils";
-import { AttachState } from "@fluidframework/container-definitions";
-import { IFluidHandle, IFluidSerializer } from "@fluidframework/core-interfaces";
+import { IFluidSerializer } from "@fluidframework/core-interfaces";
 import {
     IChannelAttributes,
     IFluidDataStoreRuntime,
-    IChannelStorageService,
-    IChannelServices,
 } from "@fluidframework/datastore-definitions";
-import { ISequencedDocumentMessage, ITree } from "@fluidframework/protocol-definitions";
-import {
-    IGarbageCollectionData,
-    ISummaryTreeWithStats,
-} from "@fluidframework/runtime-definitions";
-import { convertToSummaryTreeWithStats, FluidSerializer } from "@fluidframework/runtime-utils";
-import { ChildLogger, EventEmitterWithErrorHandling } from "@fluidframework/telemetry-utils";
-import { SharedObjectHandle } from "./handle";
-import { SummarySerializer } from "./summarySerializer";
-import { ISharedObject, ISharedObjectEvents } from "./types";
+import { ITree } from "@fluidframework/protocol-definitions";
+import { convertToSummaryTreeWithStats } from "@fluidframework/runtime-utils";
+import { ISharedObjectStateHost, SharedObjectBase } from "./sharedObjectBase";
+import { ISharedObjectEvents } from "./types";
 
 /**
- *  Base class from which all shared objects derive
+ *  Base class for shared objects with synchronous summarization
  */
 export abstract class SharedObject<TEvent extends ISharedObjectEvents = ISharedObjectEvents>
```

Comment:
It would be easier to see the diff if both SharedObject & SharedObjectBase would stay in this file initially, and future PR would be mechanical movement of code to another file